### ISSUE TITLE
Empty CognitoSignUpOptions are required when using the Amplify.Auth.signUp()

### ISSUE THREAD

**Issue by LeFrosch** [aws-amplify/amplify-flutter#292]
Only the username and password are marked with required for Amplify.Auth.signUp, but when I don’t pass an empty instance SignUpOptions I get the following error.

Invocation:
```Dart
Future<bool> register(String email, String password) async {
  final result = await Amplify.Auth.signUp(
    username: email,
    assword: password,
  );

  return result.isSignUpComplete;
}
````
Error:
```E/amplify:flutter:auth_cognito(23284): AMPLIFY_REQUEST_MALFORMED
E/amplify:flutter:auth_cognito(23284): java.lang.Exception: AMPLIFY_REQUEST_MALFORMED
E/amplify:flutter:auth_cognito(23284): 	at com.amazonaws.amplify.amplify_auth_cognito.AuthCognito.onSignUp(AuthCognito.kt:218)
E/amplify:flutter:auth_cognito(23284): 	at com.amazonaws.amplify.amplify_auth_cognito.AuthCognito.onMethodCall(AuthCognito.kt:128)
E/amplify:flutter:auth_cognito(23284): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:233)
E/amplify:flutter:auth_cognito(23284): 	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:85)
E/amplify:flutter:auth_cognito(23284): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:692)
E/amplify:flutter:auth_cognito(23284): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/amplify:flutter:auth_cognito(23284): 	at android.os.MessageQueue.next(MessageQueue.java:336)
E/amplify:flutter:auth_cognito(23284): 	at android.os.Looper.loop(Looper.java:174)
E/amplify:flutter:auth_cognito(23284): 	at android.app.ActivityThread.main(ActivityThread.java:7710)
E/amplify:flutter:auth_cognito(23284): 	at java.lang.reflect.Method.invoke(Native Method)
E/amplify:flutter:auth_cognito(23284): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
E/amplify:flutter:auth_cognito(23284): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
I/flutter (23284): Unrecognized auth error returned from platform. See logs for details
```

I personally don’t think the error message conveys that the SignUpOptions are missing and additionally the options are not marked as required. To fix this error I created an empty map and passed it to the SignUpOptions, like that:
```Dart
Future<bool> register(String email, String password) async {
  final result = await Amplify.Auth.signUp(
    username: email,
    password: password,
    options: CognitoSignUpOptions(userAttributes: <String, dynamic>{}),
  );

  return result.isSignUpComplete;
}
```

I don’t know if this is relevant but I configured the Cognito User Pool to use an email address  as the username to sign up and sign in. Also I did not select any standard attributes. 

Platform: Android
Device: OnePlus 7T (HD1903)
Android Version: 10
SDK Version: 29

Flutter Version: 1.22.5
Amplify Version: 0.0.1-dev.5

**Comment by nitinkhobragade** [comment]
Hi,

I am getting something even more weird exception when I add CognitoSignUpOptions while registration

E/flutter (23465): [ERROR:flutter/lib/ui/ui_dart_state.cc(186)] Unhandled Exception: type '_CompactLinkedHashSet<ViewState>' is not a subtype of type 'Future<Null>?'
E/flutter (23465): #0      _FutureListener.handleError (dart:async/future_impl.dart:169:20)
E/flutter (23465): #1      Future._propagateToListeners.handleError (dart:async/future_impl.dart:716:47)
E/flutter (23465): #2      Future._propagateToListeners (dart:async/future_impl.dart:737:24)
E/flutter (23465): #3      Future._completeError (dart:async/future_impl.dart:547:5)
E/flutter (23465): #4      Future._asyncCompleteError.<anonymous closure> (dart:async/future_impl.dart:603:7)
E/flutter (23465): #5      _rootRun (dart:async/zone.dart:1354:13)
E/flutter (23465): #6      _CustomZone.run (dart:async/zone.dart:1258:19)
E/flutter (23465): #7      _CustomZone.runGuarded (dart:async/zone.dart:1162:7)
E/flutter (23465): #8      _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1202:23)
E/flutter (23465): #9      _microtaskLoop (dart:async/schedule_microtask.dart:40:21)
E/flutter (23465): #10     _startMicrotaskLoop (dart:async/schedule_microtask.dart:49:5)
E/flutter (23465): 

No clues why this error. I tried some empty delayed functions just to see if the rest of the code is fine, and it is. 
If I don't provide CognitoSignUpOptions then I get following error from AWS.

E/amplify:flutter:auth_cognito(23465): AuthException
E/amplify:flutter:auth_cognito(23465): AmplifyException{message=SignUp Request malformed., cause=null, recoverySuggestion=Ensure that options map attribute is present}
 

Any help is highly appreciated.

**Comment by Jordan-Nelson** [comment]
@LeFrosch thanks for opening this issue. This should be fixed as of version 0.2.1. `signUp()` does not require options (same behavior as you mentioned), but with 0.2.1, signUp should work without them for standard email based auth.
